src/static_array.cr

```diff
@@ -62,11 +62,13 @@ struct StaticArray(T, N)
   # ```
   # StaticArray(Int32, 3).new { |i| i * 2 } # => StaticArray[0, 2, 4]
   # ```
-  def self.new(&block : Int32 -> T)
+  def self.new(& : Int32 -> T)
     array = uninitialized self
-    N.times do |i|
-      array.to_unsafe[i] = yield i
-    end
+    buf = array.to_unsafe
+    # Using a macro prevents slow compile times: https://github.com/crystal-lang/crystal/issues/2485
```

Comment:
```suggestion
    # Unroll the loop in Crystal land to avoid slow compile times in release mode. See https://github.com/crystal-lang/crystal/issues/2485#issuecomment-644416515
```